src/components/SlateArea/components/menus/HoverMenu/FocusMenu.tsx

```diff
@@ -0,0 +1,145 @@
+import { Node, Transforms } from "slate";
+import { ReactEditor, useSlate } from "slate-react";
+import React, { useEffect, useRef, useState } from "react";
+import type { FocusMenuButtonMap } from "../../../slateAreaTypes";
+import HoverMenuButton from "./HoverMenuButton";
+import { classes } from "./hoverMenu.st.css";
+import type { CustomElement } from "src/components/SlateArea/slate";
+
+/** We key the display state by node id to make the state work correctly when multiple images are
+ * added, since there is only one instance of the menu. */
+interface DisplayState {
+  [nodeId: string]: string;
+}
+
+export interface FocusMenuProps {
+  focusMenuButtons: FocusMenuButtonMap;
+}
+
+const FocusMenu = ({ focusMenuButtons }: FocusMenuProps) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const editor = useSlate();
+  const [displayState, setDisplayState] = useState<DisplayState>({});
+  const validBlockTypes = Object.keys(focusMenuButtons);
+
+  const updateMenu = () => {
+    const node = getNode();
+    const el = ref.current;
+    const { selection } = editor;
+
+    if (!el) {
+      return;
+    }
+
+    if (!selection || !node || !validBlockTypes.includes(node.type)) {
+      el.removeAttribute("style");
+      return;
+    }
+
+    /**
+     * Getting the DOM node as usual (like in the hover menu) only gives us the
+     * empty span within the void node. To solve this we get the grandparent of
+     * that node. This is a obviously a bit of a hack and might need changing
+     * later.
+     */
+    const findNativeVoidNode = () => {
+      const node = Node.get(editor, editor.selection!.anchor.path);
+      if (!node) return;
+      return ReactEditor.toDOMNode(editor, node).parentElement!.parentElement!;
+    };
+
+    const nativeNode = findNativeVoidNode();
+    if (!nativeNode) return;
+
+    const nativeRect = nativeNode.getBoundingClientRect();
+    const parentRect = nativeNode.parentElement!.getBoundingClientRect();
+
+    el.style.opacity = "1";
+    el.style.top = `${nativeRect.top - parentRect.top - el.offsetHeight / 2}px`;
+    el.style.left = `${nativeRect.left - parentRect.left}px`;
+  };
+
+  useEffect(updateMenu);
+  useEffect(() => {
+    // const scrollContainer = document.querySelector("[data-scroll-cards]")!;
+    // const handleScroll = () => window.requestAnimationFrame(updateMenu);
+    // scrollContainer.addEventListener("scroll", handleScroll);
+    // return () => scrollContainer.removeEventListener("scroll", handleScroll);
+  }, []);
```

Comment:
for an upcoming story: We'll have to think about how to handle this, as it's very tied to the CMS.